### PR TITLE
Hide Credit Transactions that are "Approved"

### PR DESCRIPTION
**Pull request by amichard**
#286 

The Credit transactions table should show the relevant records as a government user or as a fuel supplier. It shouldn't show "Approved" ones as it's a status in the historical data entry page indicating that it's ready to be committed, but not "Completed" yet. (Though, front-end will show "Approved" as "Completed" anyway). In other words, "Approved" transactions don't have value for Fuel Suppliers until they're actually committed (As "Approved" hasn't transferred credits yet)

Changelog:
- Changed how the filter works for get_organization_credit_trades
- More unit tests

### CODE REVIEW

**Comment by amichard** [comment]
[Credit Transactions Table | Hide transactions that have an Approved status in the back-end |1| [v0.2.0]](https://trello.com/c/4kgdeGFy/595-credit-transactions-table-hide-transactions-that-have-an-approved-status-in-the-back-end-1-v020)

File: backend/api/test_credit_trades.py [hunk 1]
```diff
@@ -483,3 +483,216 @@ def test_delete(self):
...
+            fair_market_value_per_credit=1,
+            zero_reason=None,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        credit_trades = CreditTradeService.get_organization_credit_trades(
+            self.gov_user.organization
+        )
+
+        self.assertNotIn(draft_credit_trade, credit_trades)
```
Comment (Sybrand) [inline backend/api/test_credit_trades.py]: I would have broken each one of these into a separate test. I would create the CreditTradeStatus and CreditTradeType needed as part of the setup, and then have a separate test for each of the assertions.

File: backend/api/test_credit_trades.py [hunk 2]
```diff
@@ -483,3 +483,216 @@ def test_delete(self):
...
+        credit_trades = CreditTradeService.get_organization_credit_trades(
+            test_organization_1
+        )
+
+        self.assertNotIn(draft_credit_trade, credit_trades)
+        self.assertIn(draft_credit_trade_from_fuel_supplier, credit_trades)
+        self.assertNotIn(submitted_credit_trade, credit_trades)
+        self.assertIn(submitted_credit_trade_as_respondent, credit_trades)
+        self.assertNotIn(approved_credit_trade, credit_trades)
+        self.assertIn(completed_credit_trade, credit_trades)
```
Comment (Sybrand) [inline backend/api/test_credit_trades.py]: Same comment as above. I don't think there's a good reason to bunch so many assertions in a single test. I think one should default to the "one assertion per test" approach unless there's a really good reason.

File: backend/api/test_credit_trades.py (hunk 1 above)
Comment (amichard) [reply backend/api/test_credit_trades.py]: The problem with breaking these into separate tests is that it increases the chance of something going wrong. 

The aim of the test is to check and make sure that certain statuses show up and certain statuses to not show up when calling a specific function.

Comment (Sybrand) [reply backend/api/test_credit_trades.py]: Sure, there's always a trade-off like that when splitting up closely related tests, if you feel the trade-off isn't worth it, then it probably isn't.

**Pull request merged**